document/src/main/java/com/yahoo/document/ReferenceDataType.java

```diff
@@ -98,4 +98,19 @@ public boolean isValueCompatible(FieldValue value) {
         ReferenceFieldValue rhs = (ReferenceFieldValue)value;
         return rhs.getDataType().equals(this);
     }
+
+    @Override
+    public int compareTo(DataType rhs) {
+        int cmp = super.compareTo(rhs);
+        return (cmp != 0)
+                ? cmp
+                : (rhs instanceof ReferenceDataType) ? targetType.compareTo(((ReferenceDataType) rhs).targetType) : 0;
```

Comment:
Consider adding parenthesis around the nested ternary expression to make precedence more obvious.

Reply:
Fixed